pkg/gatherers/workloads/workloads_info.go

```diff
@@ -175,23 +194,56 @@ func gatherWorkloadInfo(
 		info.PodCount += namespacePods.Count
 	}
 
-	workloadImageResize(info.PodCount)
+	return limitReached, info, imageCh, nil
+}
 
-	records := []record.Record{
-		{
-			Name: "config/workload_info",
-			Item: record.JSONMarshaller{Object: &info},
-		},
+func isPodTerminated(pod *corev1.Pod) bool {
+	return pod.Status.Phase == corev1.PodSucceeded ||
+		pod.Status.Phase == corev1.PodFailed
+}
+
+func podCanBeIgnored(pod *corev1.Pod) bool {
+	return pod.Status.Phase != corev1.PodRunning && pod.Status.Phase != corev1.PodPending ||
+		len(pod.Status.InitContainerStatuses) != len(pod.Spec.InitContainers) ||
+		len(pod.Status.ContainerStatuses) != len(pod.Spec.Containers)
+}
+
+func calculatePodShape(h hash.Hash, pod *corev1.Pod) (workloadPodShape, bool) {
+	var podShape workloadPodShape
+	var ok bool
+	podShape.InitContainers, ok = calculateWorkloadContainerShapes(h, pod.Spec.InitContainers, pod.Status.InitContainerStatuses)
+	if !ok {
+		return workloadPodShape{}, false
 	}
 
-	// wait for as many images as we can find to load
+	podShape.Containers, ok = calculateWorkloadContainerShapes(h, pod.Spec.Containers, pod.Status.ContainerStatuses)
+	if !ok {
+		return workloadPodShape{}, false
+	}
+
+	podShape.RestartsAlways = pod.Spec.RestartPolicy == corev1.RestartPolicyAlways
+
+	return podShape, true
+}
+
+func handleWorkloadImageInfo(
+	ctx context.Context,
+	info *workloadPods,
+	start time.Time,
+	imageCh chan string,
```

Comment:
This `imageCh` parameter is sent here only to be closed. It looks a little bit weird IMO. I guess we could close the channel before the function call.